x-pack/legacy/plugins/uptime/public/components/functional/monitor_status_bar.tsx

```diff
@@ -93,11 +102,29 @@ export const MonitorStatusBarComponent = ({ data, monitorId }: Props) => {
                 defaultMessage: 'Time since last check',
               }
             )}
-            grow={false}
+            grow={true}
           >
             {moment(new Date(timestamp).valueOf()).fromNow()}
           </EuiFlexItem>
         </EuiFlexGroup>
+        {certificateValidity && (
+          <>
+            <EuiSpacer size="s" />
+            <EuiText
+              color="subdued"
+              grow={false}
+              size="s"
+              aria-label={i18n.translate('xpack.uptime.monitorStatusBar.sslCertificateExpiry', {
+                defaultMessage: 'SSL certificate expires',
+              })}
+            >
+              {i18n.translate('xpack.uptime.monitorStatusBar.sslCertificateExpiry', {
+                defaultMessage: 'SSL certificate expires ',
+              })}
+              {moment(new Date(certificateValidity).valueOf()).fromNow()}
```

Comment:
This is just a question on my part - do you think we should add any logic to handle the case where there is a non-empty string? In theory this should _always_ be either a valid date string from ES, or an undefined value, but it might be worthwhile to add this `moment` wrapping code to the declaration above, and then we don't render anything in here if we can't get a useful value from `valueOf().fromNow()`. Perhaps extracting this to a function that returns `string | undefined`? WDYT?

Reply:
well, it's good to have front end checks as well. But main validation check should be performed while ingesting data via agent. I have refactored the component to make sure, it takes care of invalid date.